.github/workflows/test.yml

```diff
@@ -3,23 +3,31 @@ name: Build and Test
 
 on:
   push:
+    branches:
+      - main
   pull_request:
-  repository_dispatch:
-  schedule:
-     - cron: '05 5 1 * *' # <https://crontab.guru/#05_5_1_*_*> - "At 05:05 on day-of-month 1"  
+#  repository_dispatch:
+#  schedule:
+#     - cron: '05 5 1 * *' # <https://crontab.guru/#05_5_1_*_*> - "At 05:05 on day-of-month 1"
 
 
 jobs:
 
   build-and-test:
-    runs-on: windows-latest 
+    runs-on:
+#      - windows-latest
+      - self-hosted
+      - ephemeral
```

Comment:
I think we can keep them on ubuntu-latest. Last I checked its free for public repositories.

Reply:
Thanks, pushed 2d09f3e5b35a9d68d17c031ce4ffb92e5281b1eb to use ubuntu-latest